### ISSUE TITLE
AIOOB crash and deadlock with custom blueprint

### ISSUE THREAD

**Issue by wormzjl** [BluSunrize/ImmersiveEngineering#3353]
### Description of the issue:
I'm making a new issue since my crashlog is different from https://github.com/BluSunrize/ImmersiveEngineering/issues/3339

I have the following script:

```
mods.immersiveengineering.Blueprint.addRecipe("belts", <immersiveengineering:conveyor>.withTag({conveyorType: "immersiveengineering:conveyor"})*8, [<ore:plateSteel>, <ore:plateSteel>, <ore:gearSteel>]);
mods.immersiveengineering.Blueprint.addRecipe("belts", <magneticraft:conveyor_belt>*8, [<ore:plateSteel>, <ore:plateSteel>, <ore:gearSteel>]);
```

When clicking on the first button (IE belt), everything works fine, but when clicking on the second one (magneticraft belt), the game crashed.
Same crash happens when putting the blueprint inside a manual workbench.

### Crashlog:
https://paste.dimdev.org/yivanusepi.mccrash

### Versions & Modlist
ImmersiveEngineering-0.12-88

**Comment by wormzjl** [comment]
Testing another item, no crash this time, but got a deadlock
https://gist.github.com/wormzjl/9d2f47efdb260a634d365a11f874cf75

**Comment by Keyedrin** [comment]
I am also having the issue of the manual workbench locking the game when doing certain custom crafttweaker recipes. The automatic workbench works fine, and it is not every recipe nor is it every custom recipe.
The recipes that crash all have at least one output for a modded item from Vic's Modern Warfare, and I also tested recipes from Minecraft Transport Simulator that I added, but those ones work if that helps any.
